src/hooks/useLatestFunc.ts

```diff
@@ -1,15 +1,19 @@
 import { useRef, useEffect, useCallback } from 'react';
+import type { Maybe } from '../types';
 
 // https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function useLatestFunc<T extends (...args: any[]) => any>(fn: T) {
+export function useLatestFunc<T extends Maybe<(...args: any[]) => any>>(fn: T): T {
   const ref = useRef(fn);
 
   useEffect(() => {
     ref.current = fn;
   });
 
-  return useCallback((...args: Parameters<T>) => {
-    ref.current(...args);
+  const callbackFn = useCallback((...args: Parameters<NonNullable<T>>) => {
+    ref.current!(...args);
```

Comment:
Should we do that?
```suggestion
    return ref.current!(...args);
```

Reply:
Looks like none of the event handlers return anything. We can revisit if needed